sdk/storage/Azure.Storage.Blobs/src/BlockBlobWriteStream.cs

```diff
@@ -0,0 +1,99 @@
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Storage.Blobs.Models;
+using Azure.Storage.Blobs.Specialized;
+using Azure.Storage.Shared;
+
+namespace Azure.Storage.Blobs
+{
+    internal class BlockBlobWriteStream : StorageWriteStream
+    {
+        private readonly BlockBlobClient _blockBlobClient;
+        private readonly BlobRequestConditions _conditions;
+        private readonly List<string> _blockIds;
+
+        public BlockBlobWriteStream(
+            BlockBlobClient blockBlobClient,
+            long bufferSize,
+            long position,
+            BlobRequestConditions conditions,
+            IProgress<long> progressHandler) : base(
+                position,
+                bufferSize,
+                progressHandler)
+        {
+            ValidateBufferSize(bufferSize);
+            _blockBlobClient = blockBlobClient;
+            _conditions = conditions;
+            _blockIds = new List<string>();
+        }
+
+        protected override async Task AppendInternal(bool async, CancellationToken cancellationToken)
+        {
+            if (_buffer.Length > 0)
+            {
+                _buffer.Position = 0;
+
+                string blockId = StorageExtensions.GenerateBlockId(_position + _buffer.Length);
+
+                if (async)
+                {
+                    await _blockBlobClient.StageBlockAsync(
+                        base64BlockId: blockId,
+                        content: _buffer,
+                        conditions: _conditions,
+                        progressHandler: _progressHandler,
+                        cancellationToken: cancellationToken)
+                        .ConfigureAwait(false);
+                }
+                else
+                {
+                    _blockBlobClient.StageBlock(
+                        base64BlockId: blockId,
+                        content: _buffer,
+                        conditions: _conditions,
+                        progressHandler: _progressHandler,
+                        cancellationToken: cancellationToken);
+                }
+
+                _blockIds.Add(blockId);
+                _buffer.Clear();
+            }
+        }
+
+        protected override async Task FlushInternal(bool async, CancellationToken cancellationToken)
+        {
+            await AppendInternal(async, cancellationToken).ConfigureAwait(false);
+
+            await _blockBlobClient.CommitBlockListInternal(
+                base64BlockIds: _blockIds,
+                blobHttpHeaders: default,
+                metadata: default,
+                tags: default,
+                conditions: _conditions,
+                accessTier: default,
+                async: async,
+                cancellationToken: cancellationToken)
+                .ConfigureAwait(false);
+        }
```

Comment:
Do we need to clear _blocksIds here?

Reply:
No, if we clear block IDs, will will remove all previously created data.

Reply:
Ted, sorry I accidentally edited your comment when I was trying to respond.

Reply:
`BlockBlobClient.CommitBlockList()` always commits blocks in the block list as "Latest".  When I was experimenting with this a few days ago, it appears that blocks not included in the block list are removed.